internal/runtime/process.go

```diff
@@ -255,12 +255,17 @@ func (r *runState) filterParticipants(ctx context.Context, tx *ent.Tx, limit int
 	}
 
 	initializedParticipants := make([]*ent.Participant, 0)
-	for _, participant := range participants {
-		if participant.Uninitialized != nil && *participant.Uninitialized == true {
-			continue
-		}
 
-		initializedParticipants = append(initializedParticipants, participant)
+	if uninitialized == true {
+		initializedParticipants = participants
```

Comment:
This should actually be the opposite of below (could be one loop I think), we only want uninitialized participants if true. The idea is we want to select only uninitialized participants to make them go through a HIT so they are now initialized, so we want to be able to isolate them in both directions (only initialized / only uninitialized).